internal/bminventory/inventory.go

```diff
@@ -6443,15 +6443,14 @@ func (b *bareMetalInventory) GetKnownApprovedHosts(clusterId strfmt.UUID) ([]*co
 	return b.hostApi.GetKnownApprovedHosts(clusterId)
 }
 
-// In case cpu architecture is not x86_64 and platform is baremetal, we should extract openshift-baremetal-installer
+// In case cpu architecture is not x86_64, we should extract openshift-baremetal-installer
 // from x86_64 release image as there is no x86_64 openshift-baremetal-installer executable in arm image.
 // This flow does not affect the multiarch release images and is meant purely for using arm64 release image with the x86 hub.
 // Implementation of handling the multiarch images is done directly in the `oc` binary and relies on the fact that `oc adm release extract`
 // will automatically use the image matching the Hub's architecture.
 func isBaremetalBinaryFromAnotherReleaseImageRequired(cpuArchitecture, version string, platform *models.PlatformType) bool {
```

Comment:
The name of the func might be a bit confusing now... Maybe worth adding a comment about none platform clusters (i.e. that also the baremetal binary variant can be used).

Reply:
It's still the -baremetal- binary, so I think the name is still accurate.
TBH it might be less confusing than before, where the logic here said platform == baremetal and in the other location platform != none, and we were relying on the fact that no other platforms are supported on different architectures.